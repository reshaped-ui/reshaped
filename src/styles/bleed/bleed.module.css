@responsive [style*="--rs-bleed"] {
	@variable --rs-bleed 0;

	margin-left: calc(var(--rs-unit-x1) * var(--rs-bleed) * -1);
	margin-right: calc(var(--rs-unit-x1) * var(--rs-bleed) * -1);
}

@responsive .--bleed {
	@value true {
		border-left-style: none !important;
		border-right-style: none !important;
		border-radius: 0;
	}

	@value false {
		border-left-style: solid !important;
		border-right-style: solid !important;

		/* Depends on the radius css utility */
		border-radius: var(--rs-radius);
	}
}
