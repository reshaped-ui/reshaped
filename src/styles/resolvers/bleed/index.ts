import {
	responsiveClassNames,
	responsiveVariables,
	responsivePropDependency,
} from "utilities/props";
import * as T from "styles/types";
import s from "./bleed.module.css";

const getBleedStyles: T.StyleResolver<T.Bleed> = (value) => {
	if (value === undefined) return {};

	const classNames = responsiveClassNames(
		s,
		"--bleed",
		responsivePropDependency(value, (value) => typeof value === "number" && value > 0)
	);
	const variables = responsiveVariables("--rs-bleed", value);

	return { classNames: [s.root, classNames], variables };
};

export default getBleedStyles;
