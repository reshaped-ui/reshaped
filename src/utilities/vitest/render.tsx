import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
