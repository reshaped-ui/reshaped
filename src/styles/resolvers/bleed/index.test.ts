import { expect, test, describe } from "vitest";
import bleed from "./index";

describe("Styles/Bleed", () => {
	test("handles positive value", () => {
		expect(bleed(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(bleed(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(bleed()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(bleed({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
