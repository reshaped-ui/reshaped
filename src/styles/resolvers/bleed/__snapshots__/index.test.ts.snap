// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Bleed > handles 0 value 1`] = `
{
  "classNames": [
    "_root_173dfa",
    [
      "_--bleed-false_173dfa",
    ],
  ],
  "variables": {
    "--rs-bleed-s": 0,
  },
}
`;

exports[`Styles/Bleed > handles positive value 1`] = `
{
  "classNames": [
    "_root_173dfa",
    [
      "_--bleed_173dfa",
    ],
  ],
  "variables": {
    "--rs-bleed-s": 4,
  },
}
`;

exports[`Styles/Bleed > handles responsive value 1`] = `
{
  "classNames": [
    "_root_173dfa",
    [
      "_--bleed_173dfa",
      "_--bleed-false--m_173dfa",
      "_--bleed-true--l_173dfa",
    ],
  ],
  "variables": {
    "--rs-bleed-l": 2,
    "--rs-bleed-m": 0,
    "--rs-bleed-s": 4,
  },
}
`;

exports[`Styles/Bleed > handles undefined value 1`] = `{}`;
